File: sync_commands.py
import os
import re
import json

# Configuration
BOT_ROOT = './src'
WEB_ROOT = '.'
HTML_FILES = ['website/index.html', 'website/kozzyxbotmain.html']

# Categories mapping
DIR_TO_CAT = {
    'general': 'utility',
    'fun': 'fun'
}

def extract_commands():
    commands = []
    
    slash_root = os.path.join(BOT_ROOT, 'slashCommands')
    if not os.path.exists(slash_root):
        return commands

    for category in os.listdir(slash_root):
        cat_path = os.path.join(slash_root, category)
        if not os.path.isdir(cat_path):
            continue
            
        cat_label = DIR_TO_CAT.get(category, 'utility')
        
        for file in os.listdir(cat_path):
            if file.endswith('.js'):
                with open(os.path.join(cat_path, file), 'r') as f:
                    content = f.read()
                    # Simple regex to find name and description in export default
                    name_match = re.search(r'name:\s*"([^"]+)"', content)
                    desc_match = re.search(r'description:\s*"([^"]+)"', content)
                    
                    if name_match and desc_match:
                        commands.append({
                            "cmd": f"/{name_match.group(1)}",
                            "args": "",
                            "desc": desc_match.group(1),
                            "cat": cat_label,
                            "popular": False
                        })
    return commands

def update_html_files(commands):
    # Convert commands to a pretty JS array string
    js_array = "const COMMANDS = " + json.dumps(commands, indent=12) + ";"
    
    # Fix the bracket formatting slightly for aesthetic consistency
    js_array = js_array.replace('const COMMANDS = [', 'const COMMANDS = [\n            // Automatically Synced Commands')
    
    marker_begin = '/*COMMANDS-DATA-BEGIN*/'
    marker_end = '/*COMMANDS-DATA-END*/'
    
    for filename in HTML_FILES:
        filepath = os.path.join(WEB_ROOT, filename)
        if not os.path.exists(filepath):
            continue
            
        with open(filepath, 'r') as f:
            content = f.read()
            
        # Find markers
        start_idx = content.find(marker_begin)
        end_idx = content.find(marker_end)
        
        if start_idx == -1 or end_idx == -1:
            print(f"Markers not found in {filename}")
            continue
            
        # Replace content between markers
        new_content = (
            content[:start_idx + len(marker_begin)] +
            f"\n        {js_array}\n        " +
            content[end_idx:]
        )
        
        with open(filepath, 'w') as f:
            f.write(new_content)
        print(f"Updated {filename}")

if __name__ == "__main__":
    cmds = extract_commands()
    if cmds:
        update_html_files(cmds)
        print(f"Successfully synced {len(cmds)} commands.")
    else:
        print("No commands found to sync.")

File: scripts/sync_commands.js
import fs from 'fs';
import path from 'path';

const BOT_ROOT = './src';
const WEB_ROOT = '.';
const HTML_FILES = ['website/index.html', 'website/kozzyxbotmain.html'];

const DIR_TO_CAT = {
    'general': 'utility',
    'fun': 'fun'
};

function extractCommands() {
    const commands = [];
    const slashRoot = path.join(BOT_ROOT, 'slashCommands');
    
    if (!fs.existsSync(slashRoot)) return commands;

    const categories = fs.readdirSync(slashRoot);
    
    for (const category of categories) {
        const catPath = path.join(slashRoot, category);
        if (!fs.statSync(catPath).isDirectory()) continue;
        
        const catLabel = DIR_TO_CAT[category] || 'utility';
        
        const files = fs.readdirSync(catPath);
        for (const file of files) {
            if (file.endsWith('.js')) {
                const content = fs.readFileSync(path.join(catPath, file), 'utf8');
                const nameMatch = content.match(/name:\s*"([^"]+)"/);
                const descMatch = content.match(/description:\s*"([^"]+)"/);
                
                if (nameMatch && descMatch) {
                    commands.push({
                        cmd: `/${nameMatch[1]}`,
                        args: "",
                        desc: descMatch[1],
                        cat: catLabel,
                        popular: false
                    });
                }
            }
        }
    }
    return commands;
}

function updateHtmlFiles(commands) {
    let jsArray = "const COMMANDS = " + JSON.stringify(commands, null, 12) + ";";
    jsArray = jsArray.replace('const COMMANDS = [', 'const COMMANDS = [\n            // Automatically Synced Commands');
    
    const markerBegin = '/*COMMANDS-DATA-BEGIN*/';
    const markerEnd = '/*COMMANDS-DATA-END*/';
    
    for (const filename of HTML_FILES) {
        const filepath = path.join(WEB_ROOT, filename);
        if (!fs.existsSync(filepath)) continue;
        
        const content = fs.readFileSync(filepath, 'utf8');
        const startIdx = content.indexOf(markerBegin);
        const endIdx = content.indexOf(markerEnd);
        
        if (startIdx === -1 || endIdx === -1) {
            console.log(`Markers not found in ${filename}`);
            continue;
        }
        
        const newContent = content.substring(0, startIdx + markerBegin.length) +
            `\n        ${jsArray}\n        ` +
            content.substring(endIdx);
            
        fs.writeFileSync(filepath, newContent);
        console.log(`Updated ${filename}`);
    }
}

const cmds = extractCommands();
if (cmds.length > 0) {
    updateHtmlFiles(cmds);
    console.log(`Successfully synced ${cmds.length} commands.`);
} else {
    console.log("No commands found to sync.");
}
